### DISCUSSION TITLE
doctests on python fenced code blocks

### DISCUSSION THREAD

**Discussion by darikg** [mkdocstrings/mkdocstrings#660] [Q&A] [answered]
Wondering if anybody uses some variation of doctests with mkdocstrings.

I've found that I can use [sybil](https://sybil.readthedocs.io/en/latest/) to run extract fenced code blocks in docstrings to run as tests to at least ensure they run correctly, but I don't see any obvious way to assert return values as in traditional pytests doctests, e.g.

```
>>> 2 * 2
4
```

asserts that 2 * 2 == 4, but I can't figure out how to do the equivalent in a code block without explicitly `assert`ing it.

**Comment by pawamoy** [comment, marked as answer]
Sounds like a feature request for sybil :smile: 
Maybe you can implement this with their support for other languages (see https://sybil.readthedocs.io/en/latest/use.html#worked-example), here the language being `pycon` (python console) instead of `python` :shrug:

**Reply by darikg** [reply]
Sounds good :) Just wanted to make sure it wasn't a solved problem already
